woocommerce/woocommerce-android
# Issue/3725 more readers found UI

Parent issue #3725 

Merge instructions
~~1. Make sure https://github.com/woocommerce/woocommerce-android/pull/4211 is merged first~~
~~2. Remove "do not merge" label~~
3. Merge this PR

This PR adds Ui for the logic implemented in https://github.com/woocommerce/woocommerce-android/pull/4211.


Designs:
- it seems the line height used in designs is custom -> I used the default line height defined in the OS. Let me know if you'd prefer me to introduce a custom definition of line height.

| Design | Light (gif + static) | Dark (gif + static) |
|---|---|---|
| <img width="214" alt="light" src="https://user-images.githubusercontent.com/2261188/122120144-a68bf500-cdef-11eb-9363-6a0db9f3b596.png"> | <img src="https://user-images.githubusercontent.com/2261188/122118431-9f63e780-cded-11eb-8c2a-4eae74f38c25.gif" width="214px" /> | <img width="214" alt="dark" src="https://user-images.githubusercontent.com/2261188/122120464-139f8a80-cdf0-11eb-8ecf-bc6046769694.gif"> |
|  | <img width="214" alt="light static" src="https://user-images.githubusercontent.com/2261188/122120944-a4766600-cdf0-11eb-958f-cdbc69c99d7a.png"> | <img width="214" alt="dark static" src="https://user-images.githubusercontent.com/2261188/122120938-a2aca280-cdf0-11eb-9c0b-b4e28ee97d6d.png"> |








To test:
1. Open detail of an unpaid order in US dollars
2. Notice a single reader is found and UI for a single reader is shown
-----------------------------------
Prerequisite: Unfortunately, I'm not aware of an easy way how to simulate that StripeTerminalSDK found multiple readers. I manually edited [this line](https://github.com/woocommerce/woocommerce-android/blob/2d8146f946b3f54e781982ffa53d25bd5024e650/libs/cardreader/src/debug/java/com/woocommerce/android/cardreader/internal/connection/ConnectionManager.kt#L44) to return multiple readers.
1. Open detail of an unpaid order in US dollars
2. Tap on "Collect Payment"
3. Connection flow automatically starts and displays a list of found readers after a few seconds
4. Notice the "Scanning for readers" progress icon is animated
4. Tap on "Cancel" and notice the dialogs is dismissed
5. Tap on "Collect Payment"
6. Tap on one of the found readers
7. Notice the app connects to it and continues with the payment flow

Update release notes:



Update:
Issues that need to be addressed:
- [ ] Update error message on timeout - https://github.com/woocommerce/woocommerce-android/pull/4212#issuecomment-862574979
